cranelift-codegen/src/ir/constant.rs

```diff
@@ -79,7 +139,8 @@ impl ConstantPool {
     /// Insert constant data into the pool, returning a handle for later referencing; when constant
     /// data is inserted that is a duplicate of previous constant data, the existing handle will be
     /// returned.
-    pub fn insert(&mut self, constant_value: ConstantData) -> Constant {
+    pub fn insert(&mut self, constant_value: impl Into<ConstantData>) -> Constant {
```

Comment:
I have one place in `convert_ineg` where I am passing `ConstantData` directly and I was hoping to use that in more locations since it is a more specific type than `Vec`; I added the `impl Into<ConstantData>` because I didn't want to immediately change all the other call sites (not that there's that many). I sort of thought that "into from ConstantData to ConstantData" would become a no-op for the Rust compiler and that it didn't matter whether we did the "into from Vec to ConstantData" inside the `insert` or outside. I do realize that this is in the compilation path so let's do whatever is most efficient--what _is_ most efficient here?

Reply:
Ok, I thought it was passing Vec in non-testing code. As far as I know, the Rust compiler will create a copy of the function for each type of constant_value passed here, so this just grows the code's size. I think it's fine to use it here, but that's something to keep in mind. For one thing, the compilation time of the meta-crate might be very high because we have so many generic type specialization happening.

Reply:
So I guess the best thing (for code size and meta compile times) would be to convert all the call sites to use just one type, either `ConstantData` or `Vec`? I'm leaning toward `ConstantData` so that we can add to that struct in the future if necessary...

Reply:
Yeah, using only one of them sounds great; I don't have a strong opinion on which one. (My comment about the meta-crate was that the Into<> idiom is probably overly used there, resulting in higher compile times there; but this particular case wouldn't change anything about the meta crate compilation time, to be clear.)

Reply:
Ok, I'll change them to `ConstantData` in hopes that it is future-proof.